sources/api/apiserver/src/server/mod.rs

```diff
@@ -354,6 +363,43 @@ async fn get_configuration_files(
     Ok(ConfigurationFilesResponse(resp))
 }
 
+/// Get the update status from 'thar-be-updates'
+async fn get_update_status() -> Result<UpdateStatusResponse> {
+    let lockfile = File::create(UPDATE_LOCKFILE).context(error::UpdateLock)?;
+    lockfile.try_lock_shared().context(error::UpdateShareLock)?;
+    let result = thar_be_updates::status::get_update_status();
```

Comment:
Hmm, maybe `get_update_status` should take the lock as a parameter so you can't call it without locking.

Reply:
Is the suggestion for `get_update_status` to try and obtain the lock inside the function?
I'd rather keep `get_update_status` as it is right now where it just opens the file and returns the de-serialized update status struct.

Reply:
No, I'm saying that it's dangerous to call `get_update_status` unless you have the lock, so we can force the caller to have a lock by taking it as a parameter.  `get_update_status` would ignore the parameter.  (I'm not sure we can actually check that it's a lock on the same file, since fs2 just returns `()`, but it's still a strong signal to the caller.)